### PR TITLE
Rabbit ocf ra travis

### PR DESCRIPTION
**Pull request by bogdando**
Add a Travis CI job for the rabbit OCF RA cluster bootstrap and smoke test.
NOTE: it ignores changes to any files but the scripts/rabbitmq-server-ha.ocf

### CODE REVIEW

**Comment by bogdando** [comment]
@michaelklishin please take a look. I tested it for my fork and it works for me...

**Comment by bogdando** [comment]
*Note*, it relies on an external artifact DOCKER_IMAGE=bogdando/rabbitmq-cluster-ocf-wily
and Vagrantfile + Vagrant provisioning scripts from an external repo https://github.com/bogdando/rabbitmq-cluster-ocf-vagrant.git.

So, consider to either make it non voting or keep it being voting but rework the external parts to be the part of the rabbitmq-server project as well.

**Comment by bogdando** [comment]
I'm not sure how to merge the change to your current .travis.yml

The test PR (a Shall Fail example) https://github.com/bogdando/rabbitmq-server/pull/5 doesn't work for me - it fails too early :) https://travis-ci.org/bogdando/rabbitmq-server/builds/116115450

**Comment by bogdando** [comment]
Here is how an example shall pass test works for my fork if I use my own .travis.yml https://travis-ci.org/bogdando/rabbitmq-server/builds/116100594

**Comment by bogdando** [comment]
An example PR https://github.com/bogdando/rabbitmq-server/pull/5 , which Travis CI check GROUP=2 (see the 45.1 gate only):
- shall fail https://travis-ci.org/bogdando/rabbitmq-server/jobs/116131980
- shall pass (ignore as no changes to the OCF RA) https://travis-ci.org/bogdando/rabbitmq-server/builds/116134576
- shall pass (with changes made to the OCF RA) https://travis-ci.org/bogdando/rabbitmq-server/jobs/116135836

**Comment by bogdando** [comment]
@dmitrymex

File: scripts/travis_test_ocf_ra.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export DOCKER_MOUNTS="$(pwd)/scripts/rabbitmq-server-ha.ocf:/tmp/rabbitmq-server-ha"
+
+# Install vagrant and requirements
+sudo apt-get install -qq git wget
+wget --no-verbose https://releases.hashicorp.com/vagrant/${VAGRANT_VERSION}/vagrant_${VAGRANT_VERSION}_x86_64.deb
+sudo dpkg -i --force-all ./vagrant_${VAGRANT_VERSION}_x86_64.deb
+vagrant plugin install vagrant-triggers
+
+# Update docker and prepare images
+sudo apt-get update
```
Comment (dmitrymex) [inline scripts/travis_test_ocf_ra.sh]: Should not we instead do 'apt-get update' before the first install in line 15?

Comment (bogdando) [inline scripts/travis_test_ocf_ra.sh]: Makes sense, although any version of the git and wget will fit out needs, the update is mostly required for the proper docker version

Comment (dmitrymex) [inline scripts/travis_test_ocf_ra.sh]: It is a general advice I know to do update before any install. The reasoning is nicely outlined in that answer - http://askubuntu.com/a/147029

**Comment by bogdando** [comment]
No one's interested? ;(

**Comment by michaelklishin** [comment]
@bogdando we have an off-site for the entire team this week. We trust your judgement on what's the best way to test OCF.

**Comment by dmitrymex** [comment]
As for me, I am ok with the current approach. I only wish somebody who understands travis configuration to look into the PR as well to verify we do not break it

**Comment by michaelklishin** [comment]
Travis setup for rabbitmq-server is currently semi-broken: it doesn't check out the correct branch in certain cases, plus our test suite doesn't output anything for up to 1 hour sometimes. We are working on moving the test suite to Common Test and will take care of those things shortly after.

**Comment by michaelklishin** [comment]
@dumbbell assigning it to you since you're working on the test suite at the moment.

**Comment by bogdando** [comment]
Thank you! Will you enable web hooks to see job results?

**Comment by michaelklishin** [comment]
We need to revisit how we use Travis because it wasn't really created for scenarios such as ours: multiple closely related repositories.

**Pull request merged**